src/ansys/fluent/core/services/field_data.py

```diff
@@ -593,6 +593,88 @@ def _pathlines_data(
         return path_lines_dict
 
 
+class SurfaceFieldData:
+    """Container for storing parameters for surface data."""
+
+    def __init__(
+        self,
+        data_types: List[SurfaceDataType] | List[str],
+        surfaces: List[int | str],
+        overset_mesh: bool | None = False,
+    ):
+        """__init__ method of SurfaceFieldData class."""
+        self.data_types = data_types
+        self.surfaces = surfaces
+        self.overset_mesh = overset_mesh
+
+
+class ScalarFieldData:
+    """Container for storing parameters for scalar field data."""
+
+    def __init__(
+        self,
+        field_name: str,
+        surfaces: List[int | str],
+        node_value: bool | None = True,
+        boundary_value: bool | None = True,
+    ):
+        """__init__ method of ScalarFieldData class."""
+        self.field_name = field_name
+        self.surfaces = surfaces
+        self.node_value = node_value
+        self.boundary_value = boundary_value
+
+
+class VectorFieldData:
+    """Container for storing parameters for vector field data."""
+
+    def __init__(
+        self,
+        field_name: str,
+        surfaces: List[int | str],
+    ):
+        """__init__ method of VectorFieldData class."""
+        self.field_name = field_name
+        self.surfaces = surfaces
+
+
+class PathlinesFieldData:
+    """Container for storing parameters for path-lines field data."""
+
+    def __init__(
+        self,
+        field_name: str,
+        surfaces: List[int | str],
+        additional_field_name: str = "",
+        provide_particle_time_field: bool | None = False,
+        node_value: bool | None = True,
+        steps: int | None = 500,
+        step_size: float | None = 500,
+        skip: int | None = 0,
+        reverse: bool | None = False,
+        accuracy_control_on: bool | None = False,
+        tolerance: float | None = 0.001,
+        coarsen: int | None = 1,
+        velocity_domain: str | None = "all-phases",
+        zones: list | None = None,
+    ):
+        """__init__ method of PathlinesFieldData class."""
+        self.field_name = field_name
+        self.surfaces = surfaces
+        self.additional_field_name = additional_field_name
+        self.provide_particle_time_field = provide_particle_time_field
+        self.node_value = node_value
+        self.steps = steps
+        self.step_size = step_size
+        self.skip = skip
+        self.reverse = reverse
+        self.accuracy_control_on = accuracy_control_on
+        self.tolerance = tolerance
+        self.coarsen = coarsen
+        self.velocity_domain = velocity_domain
+        self.zones = zones
+
+
```

Comment:
@prmukherj What was the rationale for deciding against use of `NamedTuple`?

Reply:
Actually the other methods will cease to exist if we have this and there we can just pass **kwargs. So for documenting the args I have kept it like it is. We can decide on this.

Reply:
Thank you for the context. Will update it.